### DISCUSSION TITLE
`klog bk set`: create file if it doesn’t exist

### DISCUSSION THREAD

**Discussion by bevel-zgates** [jotaen/klog#351] [Feature ideas & discussions]
Working off of linux here, but I believe this would apply to other OSs.

It would be nice for `klog bk set ~/new_bookmark_file.klg @default` to automatically create the file for the bookmark, or if the file does not exist for you to get a prompt to let you know that a new file will be created.

Alternatively how about a `klog init` that will setup a default bookmark file for you.

**Comment by bevel-zgates** [comment]
Interesting, it looks like that is the intended behavior looking at the cli a bit here:
```
	file, err := app.NewFile(opt.File)
```
https://github.com/jotaen/klog/blob/ac17267b9f3fc780cb13f2742f93adeae3d77fdc/klog/app/cli/bookmarks.go#L99C19-L99C26

If we like the above idea I can work up a quick implementation.

OS details:

```
Distributor ID: Debian
Description:    Debian GNU/Linux 12 (bookworm)
Release:        12
Codename:       bookworm
```

**Comment by jotaen** [comment]
Currently, it is not intended for `klog bookmarks set` to create files on disk. Code-wise, the [`app.NewFile(string)` function](https://github.com/jotaen/klog/blob/ac17267b9f3fc780cb13f2742f93adeae3d77fdc/klog/app/file.go#L30-L32) is not meant to create a new file on disk, but it it’s only a constructor function to create a [`File` object](https://github.com/jotaen/klog/blob/ac17267b9f3fc780cb13f2742f93adeae3d77fdc/klog/app/file.go#L9-L11), which is just a thin abstraction to make it easier to work with file paths etc.

I think it would be useful if `klog bk set` would be able to create files. I think I wouldn’t want this to be the default behaviour, because it might be surprising for other people. An interactive prompt might be tricky because it could complicate things in the context of scripting. However, I’m thinking there could be a flag such as `klog bookmarks set --create` (or `klog bk set -c` for short). That wouldn’t be too much extra typing work, and the intent is clear and explicit.

**Comment by bevel-zgates** [comment]
Oh that would definitely fit the bill, and having it references in the `--help` would make the intent obvious.
